modules/messages/src/lanes_manager.rs

```diff
@@ -0,0 +1,280 @@
+	OutboundLaneStorage, OutboundLanes, OutboundMessages, StoredInboundLaneData,
+	StoredMessagePayload,
+};
+
+use bp_messages::{
+	ChainWithMessages, InboundLaneData, LaneId, LaneState, MessageKey, MessageNonce,
+	MessagePayload, OutboundLaneData, VerificationError,
+};
+use bp_runtime::AccountIdOf;
+use codec::MaxEncodedLen;
+use frame_support::{ensure, RuntimeDebug};
+use sp_std::marker::PhantomData;
+
+/// Lanes manager errors.
+#[derive(RuntimeDebug, PartialEq, Eq)]
+pub enum LanesManagerError {
+	/// Inbound lane already exists.
+	InboundLaneAlreadyExists,
+	/// Outbound lane already exists.
+	OutboundLaneAlreadyExists,
+	/// No inbound lane with given id.
+	UnknownInboundLane,
+	/// No outbound lane with given id.
+	UnknownOutboundLane,
+	/// Inbound lane with given id is closed.
+	ClosedInboundLane,
+	/// Outbound lane with given id is closed.
+	ClosedOutboundLane,
+}
+
+impl<T, I> From<LanesManagerError> for Error<T, I> {
+	fn from(e: LanesManagerError) -> Self {
+		match e {
+			LanesManagerError::InboundLaneAlreadyExists => Error::InboundLaneAlreadyExists,
+			LanesManagerError::OutboundLaneAlreadyExists => Error::OutboundLaneAlreadyExists,
+			LanesManagerError::UnknownInboundLane => Error::UnknownInboundLane,
+			LanesManagerError::UnknownOutboundLane => Error::UnknownOutboundLane,
+			LanesManagerError::ClosedInboundLane => Error::ClosedInboundLane,
+			LanesManagerError::ClosedOutboundLane => Error::ClosedOutboundLane,
+		}
+	}
+}
+
+/// Message lanes manager.
+pub struct LanesManager<T, I>(PhantomData<(T, I)>);
+
+impl<T: Config<I>, I: 'static> Default for LanesManager<T, I> {
+	fn default() -> Self {
+		Self::new()
+	}
+}
+
+impl<T: Config<I>, I: 'static> LanesManager<T, I> {
+	/// Create new lanes manager.
+	pub fn new() -> Self {
+		Self(PhantomData)
+	}
+
+	/// Create new inbound lane in `Opened` state.
+	pub fn create_inbound_lane(
+		&self,
+		lane_id: LaneId,
+	) -> Result<InboundLane<RuntimeInboundLaneStorage<T, I>>, LanesManagerError> {
+		InboundLanes::<T, I>::try_mutate(lane_id, |lane| match lane {
+			Some(_) => Err(LanesManagerError::InboundLaneAlreadyExists),
+			None => {
+				*lane = Some(StoredInboundLaneData(InboundLaneData {
+					state: LaneState::Opened,
+					..Default::default()
+				}));
+				Ok(())
+			},
+		})?;
+
+		self.inbound_lane(lane_id)
+	}
+
+	/// Create new outbound lane in `Opened` state.
+	pub fn create_outbound_lane(
+		&self,
+		lane_id: LaneId,
+	) -> Result<OutboundLane<RuntimeOutboundLaneStorage<T, I>>, LanesManagerError> {
+		OutboundLanes::<T, I>::try_mutate(lane_id, |lane| match lane {
+			Some(_) => Err(LanesManagerError::OutboundLaneAlreadyExists),
+			None => {
+				*lane = Some(OutboundLaneData { state: LaneState::Opened, ..Default::default() });
+				Ok(())
+			},
+		})?;
+
+		self.outbound_lane(lane_id)
+	}
+
+	/// Get existing inbound lane, checking that it is in usable state.
+	pub fn inbound_lane(
```

Comment:
1) I've renamed `inbound_lane` to `active_inbound_lane`, but left previous names for `any_state_inbound_lane`. That's because the "users" of this class will probably just see the `fn inbound_lane` and will use it without caution, which may lead to some hard-to-detect issues;
2) I've decided not to fix it - name are for methods that are visible outside of the crate and it is imo easier to understand names than some flags (compare `lanes_manager.active_inbound_lane()` vs `lanes_manager.inbound_lane(false)`). And `from_lane_id` is internal method and flag = less code lines, so imo it is better.